x-pack/plugins/observability_solution/slo/server/services/slo_repository.ts

```diff
@@ -15,10 +15,14 @@ import { SLOIdConflict, SLONotFound } from '../errors';
 import { SO_SLO_TYPE } from '../saved_objects';
 
 export interface SLORepository {
-  save(slo: SLODefinition, options?: { throwOnConflict: boolean }): Promise<SLODefinition>;
+  getExistingSavedObjectId(
+    slo: SLODefinition,
+    options?: { throwOnConflict: boolean }
+  ): Promise<string | undefined>;
+  save(slo: SLODefinition, options?: { existingSavedObjectId?: string }): Promise<SLODefinition>;
```

Comment:
I have some concerns about this change:

1. `getExistingSavedObjectId` is leaking an implementation details of the repository. I would really like to avoid surfacing the storage mechanism outside of the repository.
2. Having `getExistingSavedObjectId`  and `throwOnConflict` is slightly confusing. Why would I want to throw on conflict if I want to get the **existing** internal storage id. 


Overall I think this particular optimization brings more complexity while breaking the principle of encapsulation of a Repository. I think we get 80% of the good stuff with just the other change done in the execution of the promises.